### PR TITLE
fix: productdetailpage color variant selector

### PR DESCRIPTION
**Pull request by feliciakri**
Task: https://phabricator.sirclo.com/T36910

The reason for the previously broken variant selector on prod was because the variant id was naively selected through SCSS so it only handles variants with ID of "size" and "warna". This issue has been fixed with this PR. 
image

Another issue is it is not possible to differentiate variants type. There exists two types of variants on prius
image

So the color variant is selected naively through ID on CSS. This is a bug found in other templates too (tested onmelodika, merlin). Not yet tested on other templates.
Screenshot from 2022-08-26 14-44-12
Screenshot from 2022-08-26 14-44-59

This is because the color variant is hardcoded through nexus
image

that results in a jsx element like so:
image (1)

On this PR only color type variant with ID "colors" is handled meant to be temporary fix. 

This PR variant selection:

Figma: 
https://www.figma.com/file/AMtQECMzDn4EXf4geUmzUI/%5BVENDOR%5D-Framic?node-id=29%3A1363

Mobile
https://user-images.githubusercontent.com/8079082/186859878-6c247e1c-d3ef-4ef5-ab25-a211497f9165.mp4

Desktop
https://user-images.githubusercontent.com/8079082/186859884-7a60f2ee-c67c-4ff6-834e-245fff3b48e3.mp4

### CODE REVIEW

File: public/scss/components/ProductDetail.module.scss [hunk 1]
```diff
@@ -327,15 +325,17 @@
...
     }
 
-    &[id="warna"]
+    &[id="colors"]
     {
       & > div[class*="productdetail_variantOption"]
       {
         & > span
         {
+          margin: 1rem 1rem 0.5rem 0.5rem;
```
Comment (falmesino) [inline public/scss/components/ProductDetail.module.scss:337]: ganti pake px aja nyonya

**Review by falmesino — CHANGES_REQUESTED** [review]
margin yang pake rem ganti ke px aja nyonya,

https://medium.com/@sascha.wolff/dont-use-rem-em-for-paddings-margins-and-more-94e19026b000

**Review by falmesino — APPROVED** [review]

File: public/scss/components/ProductDetail.module.scss [hunk 2]
```diff
@@ -359,13 +358,23 @@
...
-            border: 1px solid $color_light;
             box-sizing: border-box;
-            padding: 12px;
-            border-radius: 2px;
+            -moz-box-sizing: border-box;
+            -webkit-box-sizing: border-box;
+            outline: 1px solid transparent;
+            border-radius: 50%;
+            border: 3px solid $color_white;
+            width: 40px;
```
Comment (mayun19) [inline public/scss/components/ProductDetail.module.scss:367]: mba fel, boleh request pake mixin ?

**Review by mayun19 — CHANGES_REQUESTED** [review]

**Review by mayun19 — APPROVED** [review]

File: public/scss/components/ProductDetail.module.scss [hunk 3]
```diff
@@ -289,10 +289,8 @@
   {
     margin-bottom: 34px;
 
-    &[id="size"]
+    & > div[class*="productdetail_variantOption"]
     {
-      & > div[class*="productdetail_variantOption"]
-      {
         & > span
```
Comment (sultanfariz) [inline public/scss/components/ProductDetail.module.scss:294]: kak kayanya ini kelebihan indentasi sedikit sama di bawah scopenya kelebihan enter jugaa

**Review by sultanfariz — CHANGES_REQUESTED** [review]

File: public/scss/components/ProductDetail.module.scss (hunk 3 above)
Comment (sultanfariz) [inline public/scss/components/ProductDetail.module.scss:294]: kak kayanya ini kelebihan indentasi sedikit sama di bawah scopenya kelebihan enter jugaa

Comment (sultanfariz) [inline public/scss/components/ProductDetail.module.scss:294]: kak kayanya ini kelebihan indentasi sedikit sama di bawah scopenya kelebihan enter jugaa

**Review by sultanfariz — CHANGES_REQUESTED** [review]

File: public/scss/components/ProductDetail.module.scss [hunk 4]
```diff
@@ -327,15 +325,17 @@
             cursor: pointer;
           }
         }
-      }
+
```
Comment (inezamandha) [inline public/scss/components/ProductDetail.module.scss:328]: yang ini maksudnya sultan kelebihan enter ya

**Comment by inezamandha** [comment]
boleh request ss nya tidak hasil yang di sini bagaimana mba?

Related Product dan warna warnanyaa

**Review by inezamandha — APPROVED** [review]

**Review by sultanfariz — APPROVED** [review]

**Pull request merged**